src/Settings/Builders/Haddock.hs

```diff
@@ -1,25 +1,26 @@
 module Settings.Builders.Haddock (haddockBuilderArgs) where
 
-import Hadrian.Utilities
 import Hadrian.Haskell.Cabal
-
+import Hadrian.Haskell.Cabal.Configured as ConfCabal
+import Hadrian.Utilities
 import Rules.Documentation
 import Settings.Builders.Common
 import Settings.Builders.Ghc
 
 -- | Given a version string such as "2.16.2" produce an integer equivalent.
 versionToInt :: String -> Int
-versionToInt s = case map read . words $ replaceEq '.' ' ' s of
-    [major, minor, patch] -> major * 1000 + minor * 10 + patch
-    _                     -> error "versionToInt: cannot parse version."
+versionToInt = read . dropWhile (=='0') . filter (/='.')
```

Comment:
Why deviate from the Make build system in this aspect? See:

https://github.com/ghc/ghc/blob/master/rules/haddock.mk#L18

Reply:
IIRC I hit the `error` branch that this code deletes, I can try reverting this change and seeing if that still happens when I get a chance to.